src/state/ledger/mod.rs

```diff
@@ -327,8 +327,10 @@ mod tests {
             .apply_diff(&ledger_diff)
             .expect("ledger diff application");
 
+        let account_before = account.clone();
         let account_after = ledger.accounts.get(&public_key).expect("account get");
 
+        assert_eq!(account_before.balance, Amount(10)); // with cloning
```

Comment:
Instead of having two asserts with hardcoded values, make it a single assert
```rust
assert_eq!(account_after, account_before.add(ledger_diff.account_diffs[0].amount);
```

(you will also need to implement the `add()` method on `Amount`)

Reply:
This required additional code so I'm adding it and will push it here soon for you to see.